src/main/java/org/jabref/gui/actions/ExportLinkedFilesAction.java

```diff
@@ -0,0 +1,169 @@
+    private long totalFilesCount;
+    private BibDatabaseContext databaseContext;
+    private Optional<Path> exportPath = Optional.empty();
+    private List<BibEntry> entries;
+
+    public ExportLinkedFilesAction() {
+        super(Localization.lang("Copy attached files to folder..."));
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+
+        DirectoryDialogConfiguration dirDialogConfiguration = new DirectoryDialogConfiguration.Builder()
+                .withInitialDirectory(Paths.get(Globals.prefs.get(JabRefPreferences.EXPORT_WORKING_DIRECTORY)))
+                .build();
+        entries = JabRefGUI.getMainFrame().getCurrentBasePanel().getSelectedEntries();
+        exportPath = DefaultTaskExecutor
+                .runInJavaFXThread(() -> ds.showDirectorySelectionDialog(dirDialogConfiguration));
+
+        exportPath.ifPresent(path -> {
+            databaseContext = JabRefGUI.getMainFrame().getCurrentBasePanel().getDatabaseContext();
+            totalFilesCount = entries.stream().flatMap(entry -> entry.getFiles().stream()).count();
+
+            Service<Void> exportService = new ExportService();
+            startServiceAndshowProgessDialog(exportService);
+        });
+    }
+
+    private <V> void startServiceAndshowProgessDialog(Service<V> service) {
+        DefaultTaskExecutor.runInJavaFXThread(() -> {
+            service.start();
+
+            ProgressDialog progressDialog = new ProgressDialog(service);
+            progressDialog.setOnCloseRequest(evt -> service.cancel());
+            DialogPane dialogPane = progressDialog.getDialogPane();
+            dialogPane.getButtonTypes().add(ButtonType.CANCEL);
+            Button cancelButton = (Button) dialogPane.lookupButton(ButtonType.CANCEL);
+            cancelButton.setOnAction(evt -> {
+                service.cancel();
+                progressDialog.close();
+            });
+            progressDialog.showAndWait();
+        });
+    }
+
+    private class ExportService extends Service<Void> {
+
+        private static final String LOGFILE = "exportLog.log";
+        private final String localizedSucessMessage = Localization.lang("Copied file successfully");
+        private final String localizedErrorMessage = Localization.lang("Could not copy file") + ": " + Localization.lang("File exists");
+
+        @Override
+        protected Task<Void> createTask() {
+            return new Task<Void>() {
+
+                int totalFilesCounter;
+                int numberSucessful;
+                int numberError;
+                Optional<Path> newPath;
+
+                @Override
+                protected Void call()
+                        throws InterruptedException, IOException {
+                    updateMessage(Localization.lang("Copying files..."));
+                    updateProgress(0, totalFilesCount);
+
+                    try (BufferedWriter bw = Files.newBufferedWriter(exportPath.get().resolve(LOGFILE), StandardCharsets.UTF_8)) {
```

Comment:
In my opinion, a log file is an inconvenient solution for the user. Why don't log all the errors to an internal list and display this list at the end in a new dialog?